dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/h12/grpc/GrpcHttp2ServerTransportListener.java

```diff
@@ -145,39 +146,31 @@ public void onMessage(InputStream inputStream) {
 
     private class DetermineMethodDescriptorListener implements StreamingDecoder.FragmentListener {
 
-        @Override
-        public void onFragmentMessage(InputStream rawMessage) {}
-
         @Override
         public void onClose() {
             getStreamingDecoder().close();
         }
 
         @Override
-        public void onFragmentMessage(InputStream dataHeader, InputStream rawMessage) {
+        public void onFragmentMessage(InputStream rawMessage) {
             try {
-                ByteArrayOutputStream merged =
-                        new ByteArrayOutputStream(dataHeader.available() + rawMessage.available());
-                StreamUtils.copy(dataHeader, merged);
-                byte[] data = StreamUtils.readBytes(rawMessage);
-
                 RpcInvocationBuildContext context = getContext();
                 if (null == context.getMethodDescriptor()) {
-                    context.setMethodDescriptor(DescriptorUtils.findTripleMethodDescriptor(
-                            context.getServiceDescriptor(), context.getMethodName(), data));
+                    byte[] data = StreamUtils.readBytes(rawMessage);
+                    MethodDescriptor methodDescriptor = DescriptorUtils.findTripleMethodDescriptor(
+                            context.getServiceDescriptor(), context.getMethodName(), data);
+                    context.setMethodDescriptor(methodDescriptor);
```

Comment:
**Security Issue**: The method 'DescriptorUtils.findTripleMethodDescriptor' is being used without validating the input, which could lead to injection vulnerabilities if the input is controlled by an attacker. <br> **Fix**: Ensure that 'context.getMethodName()' and 'data' are properly validated or sanitized before they are used in 'DescriptorUtils.findTripleMethodDescriptor'. <br> **Code Suggestion**: 
 ```
 Ensure that 'context.getMethodName()' and 'data' are properly validated or sanitized before they are used in 'DescriptorUtils.findTripleMethodDescriptor'.

 +                byte[] data = StreamUtils.readBytes(rawMessage);
 +                if(isValidData(data) && isValidMethodName(context.getMethodName())) {
 +                    MethodDescriptor methodDescriptor = DescriptorUtils.findTripleMethodDescriptor(
 +                                context.getServiceDescriptor(), context.getMethodName(), data);
 +                    context.setMethodDescriptor(methodDescriptor);
 +                }
 ```